src/libs/CardUtils.ts

```diff
@@ -352,6 +353,16 @@ function getSelectedFeed(lastSelectedFeed: OnyxEntry<CompanyCardFeed>, cardFeeds
     return lastSelectedFeed ?? defaultFeed;
 }
 
+function getFilteredCardList(list?: WorkspaceCardsList) {
+    const {cardList, ...cards} = list ?? {};
+    // We need to filter out cards which already has been assigned
+    return Object.fromEntries(Object.entries(cardList ?? {}).filter(([cardNumber]) => !Object.values(cards).find((card) => card.lastFourPAN && cardNumber.endsWith(card.lastFourPAN))));
+}
+
+function hasOnlyOneCardToAssign(list: FilteredCardList) {
+    return Object.keys(list).length === 1;
+}
```

Comment:
I think we don't need a util function for this logic because it is so simple and short

Reply:
yeah, but it's easier for testing if you have multiple cards setup

Reply:
and probably we would need it somewhere else